utils/src/main/scala/za/co/absa/hermes/utils/HelperFunctions.scala

```diff
@@ -72,9 +76,11 @@ object HelperFunctions {
     }
 
     def flattenNestedArrays(path: String, fieldNamePrefix: String, arrayType: ArrayType): Unit = {
-      val maxIndexes = df.agg(max(expr(s"size($path)"))).collect()(0)(0).toString.toInt
+      val maxLengths = (Seq(df) ++ Seq(dfsForArraysLength: _*))
+        .map(_.agg(max(expr(s"size($path)"))).collect()(0)(0).toString.toInt)
```

Comment:
Ditto regarding the map function. It can actually be a real function shared by both `flattenStructArray` and `flattenNestedArrays`

Reply:
Yeah, good point. Done